appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmRuntimeFileLogHandler.java

```diff
@@ -102,4 +111,57 @@ private static void reloadLoggingProperties(LogManager logManager) {
       System.err.println(e.getClass().getName() + ": " + e.getMessage());
     }
   }
+
+  @Override
+  public void close() throws SecurityException {
+    super.close();
+    if (handlers.decrementAndGet() <= 0 ) {
+      queue.offer(null);
+    }
+  }
+
+  @Override
+  public void publish(LogRecord record) {
+    if (isLoggable(record)) {
+      queue.offer(new LogEntry(record));
+    }
+  }
+
+  private void superPublish(LogRecord record) {
+    super.publish(record);
+  }
+
+  private class LogEntry {
```

Comment:
The LogEntry class is a nest class and thus holds a reference back to the handler instance that created it.    Multiple handler instances can exist and they share the queue and thread.  The async thread when doing the actual logging calls back to the handler instance.

This design is copied from the Juli log.   It may not be necessary as we "know" that there should only be a single instance of the handler.

The other option is to have a none static queue and thread, so that every instance of the handler would create their own queue.  This would not be an issue as we only create 1, but would at least work if more were created.